pkg/api/v1/orchestrator/routes/handlers.go

```diff
@@ -0,0 +1,561 @@
+	}
+}
+
+// @Summary ConditionQueuePop
+// @Tag Conditions
+// @Description Pops a conditions from the NATS Jestream and registers the serverID as a controller.
+// @ID conditionQueue
+// @Param uuid path string true "Server ID"
+// @Param conditionKind path string true "Condition Kind"
+// @Accept json
+// @Produce json
+// @Success 200 {object} v1types.ServerResponse
+// Failure 400 {object} v1types.ServerResponse
+// Failure 404 {object} v1types.ServerResponse
+// Failure 500 {object} v1types.ServerResponse
+// @Router /servers/{uuid}/condition-queue/{conditionKind} [get]
+func (r *Routes) conditionQueuePop(c *gin.Context) (int, *v1types.ServerResponse) {
+	ctx, span := otel.Tracer(pkgName).Start(c.Request.Context(), "Routes.conditionQueuePop")
+	span.SetAttributes(
+		attribute.KeyValue{Key: "serverId", Value: attribute.StringValue(c.Param("uuid"))},
+		attribute.KeyValue{Key: "conditionKind", Value: attribute.StringValue(c.Param("conditionKind"))},
+	)
+	defer span.End()
+
+	serverID, err := uuid.Parse(c.Param("uuid"))
+	if err != nil {
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "invalid server id: " + err.Error(),
+		}
+	}
+
+	conditionKind := rctypes.Kind(c.Param("conditionKind"))
+	if !r.conditionKindValid(conditionKind) {
+		r.logger.WithFields(logrus.Fields{
+			"kind": conditionKind,
+		}).Info("unsupported condition kind")
+
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "unsupported condition kind: " + string(conditionKind),
+		}
+	}
+
+	conditionRecord, err := r.repository.Get(ctx, serverID)
+	if err != nil {
+		if errors.Is(err, store.ErrConditionNotFound) {
+			return http.StatusNotFound, &v1types.ServerResponse{
+				Message: "condition not found for server",
+			}
+		}
+
+		r.logger.WithError(err).Info("condition record query error")
+
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "condition lookup: " + err.Error(),
+		}
+	}
+
+	var activeConditionExists bool
+	for _, cond := range conditionRecord.Conditions {
+		if cond.Kind == conditionKind && !rctypes.StateIsComplete(cond.State) {
+			activeConditionExists = true
+		}
+	}
+
+	if !activeConditionExists {
+		return http.StatusNotFound, &v1types.ServerResponse{
+			Message: "no active condition found for server",
+		}
+	}
+
+	// TODO: if a task exists for this condition in an active state, don't proceed
+
+	// pop condition from the Jetstream queue
+	cond, err := r.conditionJetstream.pop(ctx, conditionKind, serverID)
+	if err != nil {
+		if errors.Is(err, errNoConditionInQueue) {
+			return http.StatusNotFound, &v1types.ServerResponse{
+				Message: "no condition in queue",
+			}
+		}
+
+		r.logger.WithError(err).Info("condition queue fetch error")
+
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "Condition Queue fetch error error: " + err.Error(),
+		}
+	}
+
+	// register this worker
+	controllerID, err := r.livenessKV.register(serverID.String())
```

Comment:
The idea here was to lock the Task to the Controller ID that requested it initially, since then we've had discussions around if that is necessary at all. 
Although, now I question why register liveness at all if these "remote" controllerID's are going to be ephermeral. Without a fixed controllerID - it'll be unclear if a controller is dead or gone for a reboot, and we'd just rely on the Condition threshold to determine staleness.

Reply:
liveness is not going to be used by inband controllers, recent commit purges all liveness related code.